services/src/main/java/org/fao/geonet/api/records/MetadataApi.java

```diff
@@ -690,13 +694,84 @@ public FeatureResponse getFeatureCatalog(
+        @ApiResponse(responseCode = "403", description = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW)
+    })
+    public ResponseEntity<Boolean> checkMetadataTitleDuplicated(
+        @Parameter(description = API_PARAM_RECORD_UUID,
+            required = true)
+        @PathVariable
+        String metadataUuid,
+        @Parameter(description = "Metadata title to check",
+            required = true)
+        @RequestBody String title,
+        HttpServletRequest request
+    )
+        throws Exception {
+        try {
+            ApiUtils.canViewRecord(metadataUuid, request);
+        } catch (SecurityException e) {
+            Log.debug(API.LOG_MODULE_NAME, e.getMessage(), e);
+            throw new NotAllowedException(ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW);
+        }
+
+        Set<String> uuidsWithSameTitle = retrieveMetadataUuidsFromTitle(esSearchManager, title, metadataUuid);
+
+        return ResponseEntity.ok(!uuidsWithSameTitle.isEmpty());
+    }
 
     private boolean isIncludedAttributeTable(RelatedResponse.Fcat fcat) {
         return fcat != null
             && fcat.getItem() != null
-            && fcat.getItem().size() > 0
+            && !fcat.getItem().isEmpty()
             && fcat.getItem().get(0).getFeatureType() != null
             && fcat.getItem().get(0).getFeatureType().getAttributeTable() != null
             && fcat.getItem().get(0).getFeatureType().getAttributeTable().getElement() != null;
     }
+
+
+    /**
+     * Retrieves the list of metadata uuids that have the same dataset identifier.
+     *
+     * @param searchMan             ES search manager.
+     * @param metadataTitle         Metadata title to check.
+     * @param metadataUuidToExclude Metadata identifier to exclude from the search.
+     * @return  A list of metadata uuids that have the same metadata title.
+     */
+    private Set<String> retrieveMetadataUuidsFromTitle(EsSearchManager searchMan,
+                                                            String metadataTitle,
+                                                            String metadataUuidToExclude) {
+
+        Set<String> metadataUuids = new HashSet<>();
+
+        String jsonQuery = " {" +
+            "       \"query_string\": {" +
+            "       \"query\": \"+resourceTitleObject.\\\\*.keyword:\\\"%s\\\" -uuid:\\\"%s\\\"\"" +
+            "       }" +
+            "}";
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            JsonNode esJsonQuery = objectMapper.readTree(String.format(jsonQuery, metadataTitle, metadataUuidToExclude));
+
+            final SearchResponse queryResult = searchMan.query(
+                esJsonQuery,
+                FIELDLIST_UUID,
+                0, 1000);
+
+            for (SearchHit hit : queryResult.getHits()) {
+                String uuid = hit.getSourceAsMap().get(Geonet.IndexFieldNames.UUID).toString();
+                metadataUuids.add(uuid);
+            }
```

Comment:
Since we are only want to know if there are metadata records with the same title what is the purpose of collecting all the UUIDs? Wouldn't be more efficient to limit the query to only 1 record and check the size of the SearchHit set?

Reply:
You are right, that would be more efficient.